builds/e2e/templates/e2e-setup.yaml

```diff
@@ -0,0 +1,140 @@
+    }
+  displayName: Locate edgelet packages, runtime images
+
+- task: DownloadBuildArtifacts@0
+  displayName: Download edgelet packages
+  inputs:
+    buildType: specific
+    project: one
+    pipeline: $(az.pipeline.packages)
+    buildVersionToDownload: $(whichPackageBuild)
+    branchName: refs/heads/master
+    buildId: $(packageBuildId)
+    downloadType: specific
+    itemPattern: $(artifactDir)/*
+
+- task: DownloadBuildArtifacts@0
+  displayName: Get Docker image info
+  inputs:
+    buildType: specific
+    project: one
+    pipeline: $(az.pipeline.images)
+    buildVersionToDownload: $(whichImageBuild)
+    branchName: refs/heads/master
+    buildId: $(imageBuildId)
+    downloadType: specific
+    itemPattern: $(az.pipeline.images.artifacts)/artifactInfo.txt
+
+- pwsh: |
+    $certsDir = '$(System.ArtifactsDirectory)/certs'
+    New-Item "$certsDir" -ItemType Directory -Force | Out-Null
+    $env:ROOT_CERT | Out-File -Encoding Utf8 "$certsDir/rsa_root_ca.cert.pem"
+    $env:ROOT_KEY | Out-File -Encoding Utf8 "$certsDir/rsa_root_ca.key.pem"
+    Write-Output "##vso[task.setvariable variable=certsDir]$certsDir"
+  displayName: Install CA keys
+  env:
+    ROOT_CERT: $(TestRootCaCertificate)
+    ROOT_KEY: $(TestRootCaKey)
+
+- pwsh: |
+    $testDir = '$(Build.SourcesDirectory)/test/Microsoft.Azure.Devices.Edge.Test'
+    dotnet build $testDir
+
+    $binDir = Convert-Path "$testDir/bin/Debug/netcoreapp2.1"
+    Write-Output "##vso[task.setvariable variable=binDir]$binDir"
+  displayName: Build tests
+
+- pwsh: |
+    $imagePrefix = '$(cr.address)/$(cr.labelPrefix)azureiotedge'
+    $imageId = Get-Content -Encoding Utf8 `
+      '$(System.ArtifactsDirectory)/$(az.pipeline.images.artifacts)/artifactInfo.txt'
+    $imageId = ($imageId -split '=')[1]
+    $imageTag = "$imageId-$(os)-$(arch)"
+    $context = @{
+      edgeAgentImage = "$imagePrefix-agent:$imageTag";
+      edgeHubImage = "$imagePrefix-hub:$imageTag";
+      tempSensorImage = "$imagePrefix-simulated-temperature-sensor:$imageTag";
+      methodSenderImage = "$imagePrefix-direct-method-sender:$imageTag";
+      methodReceiverImage = "$imagePrefix-direct-method-receiver:$imageTag";
+      registries = @(
+        @{
+          address = '$(cr.address)';
+          username = '$(cr.username)';
+        }
+      );
+      packagePath = Convert-Path '$(System.ArtifactsDirectory)/$(artifactDir)';
+      caCertScriptPath = Convert-Path '$(Build.SourcesDirectory)/tools/CACertificates';
+      rootCaCertificatePath = Convert-Path '$(certsDir)/rsa_root_ca.cert.pem';
+      rootCaPrivateKeyPath = Convert-Path '$(certsDir)/rsa_root_ca.key.pem';
+      logFile = Join-Path '$(binDir)' 'testoutput.log';
+    }
+
+    if ($IsWindows)
+    {
+      $context['installerPath'] = Convert-Path '$(Build.SourcesDirectory)/scripts/windows/setup'
+    }
+
+    if ('$(arch)' -eq 'arm32v7')
+    {
+      $context['optimizeForPerformance'] = 'false'
+      $context['setupTimeoutMinutes'] = 8
```

Comment:
Should some of these be configurable by pipeline variables? for example, it is easy to update variable in vsts pipeline to increase test timeout.

Reply:
They certainly could be, but I don't see the need to do that here (yet). If, for example, we had different kinds of arm32v7 agent hardware--some slower and some faster--then it could be a good idea.